system_metrics_collector/src/topic_statistics_collector/subscriber_topic_statistics.cpp

```diff
@@ -0,0 +1,271 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <limits>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rcl/time.h"
+#include "rclcpp/rclcpp.hpp"
+#include "rcpputils/asserts.hpp"
+
+#include "received_message_age.hpp"
+#include "received_message_period.hpp"
+#include "subscriber_topic_statistics.hpp"
+#include "system_metrics_collector/constants.hpp"
+
+using metrics_statistics_msgs::msg::MetricsMessage;
+
+namespace topic_statistics_collector
+{
+static rcl_interfaces::msg::IntegerRange buildPositiveIntegerRange(
```

Comment:
While this is correct, a more modern way of defining internal linkage functions is to put them in an anonymous namespace. This can be within `topic_statistics_collector`. So something like:

```c++
namespace topic_statistics_collector
{
  namespace
  {
    rcl_interfaces::msg::IntegerRange buildPositiveIntegerRange
```